src/test/java/com/bobocode/hoverla/bring/context/ApplicationContextImplTest.java

```diff
@@ -0,0 +1,152 @@
+package com.bobocode.hoverla.bring.context;
+
+import com.bobocode.hoverla.bring.exception.NoSuchBeanException;
+import com.bobocode.hoverla.bring.exception.NoUniqueBeanException;
+import com.google.common.collect.Table;
+import org.junit.jupiter.api.*;
+import org.mockito.Mockito;
+
+import java.lang.reflect.Field;
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@TestInstance(value = TestInstance.Lifecycle.PER_CLASS)
+public class ApplicationContextImplTest {
+
+    private ApplicationContext applicationContext;
+    private BeanScanner beanScannerOne;
+    private BeanScanner beanScannerTwo;
+    private BeanValidator validator;
+    private BeanInitializer initializer;
+
+    @BeforeAll
+    private void init() {
+        beanScannerOne = Mockito.mock(BeanScanner.class);
+        beanScannerTwo = Mockito.mock(BeanScanner.class);
+        List<BeanScanner> beanScannerList = Arrays.asList(beanScannerOne, beanScannerTwo);
+
+        validator = Mockito.mock(BeanValidator.class);
+
+        initializer = Mockito.mock(BeanInitializer.class);
+
+        BeanDefinition beanDefinitionStringOne = Mockito.mock(BeanDefinition.class);
+        BeanDefinition beanDefinitionStringTwo = Mockito.mock(BeanDefinition.class);
+        BeanDefinition beanDefinitionInteger = Mockito.mock(BeanDefinition.class);
+
+        when(beanScannerOne.scan()).thenReturn(Arrays.asList(beanDefinitionStringOne, beanDefinitionInteger));
+        when(beanScannerTwo.scan()).thenReturn(Collections.singletonList(beanDefinitionStringTwo));
+
+        when(beanDefinitionStringOne.name()).thenReturn("String");
+        doReturn(String.class).when(beanDefinitionStringOne).type();
+        when(beanDefinitionStringOne.instance()).thenReturn("String bean");
+
+        when(beanDefinitionInteger.name()).thenReturn("Integer");
+        doReturn(Integer.class).when(beanDefinitionInteger).type();
+        when(beanDefinitionInteger.instance()).thenReturn(2);
+
+        when(beanDefinitionStringTwo.name()).thenReturn("One more string bean");
+        doReturn(String.class).when(beanDefinitionStringTwo).type();
+        when(beanDefinitionStringTwo.instance()).thenReturn("String bean");
+
+        applicationContext = new ApplicationContextImpl(beanScannerList, validator, initializer);
+    }
+
+    @Test
+    @DisplayName("Application context initializing")
+    public void applicationContextInitializingTest() throws NoSuchFieldException, IllegalAccessException {
+        verify(beanScannerOne).scan();
+        verify(beanScannerTwo).scan();
+
+        verify(validator).validate(anyList());
+
+        verify(initializer).initialize(any());
+
+        Field contextField = applicationContext.getClass().getDeclaredField("context");
```

Comment:
I don't think we have to use reflection here, it's a bad practice. You can just leave verifies here and check the actual context size and values in other cases

Reply:
@DanielChanger I'm not sure I understood you well. Do you propose to remove this code or to remove and add one more test method that will check size and values?